ceph/rados/pool_workflows.py

```diff
@@ -115,13 +117,65 @@ def fill_omap_entries(self, pool_name, **kwargs):
         # removing the py file copied
         client_node.exec_command(sudo=True, cmd="rm -rf generate_omap_entries.py")
 
-        # restarting primary osd as metadata does not get updated automatically
-        # when number of object containing omap entries is below ~40
-        # this a workaround and will be addressed when RCA is complete
-        primary_osd = self.rados_obj.get_pg_acting_set(pool_name=pool_name)[0]
-        self.rados_obj.change_osd_state(action="restart", target=primary_osd)
+        # Storing the pg dump log before setting the scrub parameters
+        before_scrub_log = self.scrub_obj.get_pg_dump("pgid", "last_scrub_stamp")
+        before_deep_scrub_log = self.scrub_obj.get_pg_dump(
+            "pgid", "last_deep_scrub_stamp"
+        )
+
+        # Triggering deep scrub on the pool
+        self.rados_obj.run_deep_scrub(pool=pool_name)
+
+        flag = 1
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=3000)
+        while end_time > datetime.datetime.now():
+            after_scrub_log = self.scrub_obj.get_pg_dump("pgid", "last_scrub_stamp")
+            after_deep_scrub_log = self.scrub_obj.get_pg_dump(
+                "pgid", "last_deep_scrub_stamp"
+            )
+            scrub_status = self.scrub_obj.verify_scrub_deepscrub(
+                before_scrub_log, after_scrub_log, "scrub"
+            )
+            deep_scrub_status = self.scrub_obj.verify_scrub_deepscrub(
+                before_deep_scrub_log, after_deep_scrub_log, "deepscrub"
+            )
+            if not (scrub_status & deep_scrub_status):
+                log.debug(
+                    "Scrubbing/ deep scrubbing has not been completed on the pool"
+                    "Sleeping for 10 seconds and checking again"
+                )
```

Comment:
@pdhiran I guess we need to remove the check for normal scrub as we are no longer performing normal scrub anymore